src/main/scala/is/hail/annotations/aggregators/RegionValueTakeByIntAggregator.scala

```diff
@@ -0,0 +1,50 @@
+package is.hail.annotations.aggregators
+
+import is.hail.annotations._
+import is.hail.expr.types.Type
+
+import scala.collection.mutable
+
+class RegionValueTakeByIntIntAggregator(n: Int, val inputType: Type, val keyType: Type) extends RegionValueAggregator {
+  val ord = keyType.ordering.toOrdering.on[(Any, Any)] { case (e, k) => k }
+  var _state = new mutable.PriorityQueue[(Any, Any)]()(ord)
+
+  def seqOp(p: (Any, Any)) {
+    if (_state.length < n)
+      _state += p
+    else {
+      if (ord.compare(p, _state.head) < 0) {
+        _state.dequeue()
+        _state += p
+      }
+    }
+  }
+
+  def seqOp(region: Region, x: Int, xm: Boolean, k: Int, km: Boolean) {
+    if (!xm && !km) {
```

Comment:
I think takeBy should be like sets, the should be able to contain missing elements.  So I think this can read: `seqOp(if (xm) null else x, if (km) null else k)`.